src/state/page-models/actions.js

```diff
@@ -142,45 +151,41 @@ export const initPageModelForm = () => (dispatch, getState) => {
   });
 };
 
-export const updatePageModel = pageModel => (dispatch, getState) => {
-  const toSend = pageModel || getFormPageModel(getState());
-  return new Promise((resolve) => {
-    if (toSend) {
-      putPageModel(toSend).then((response) => {
-        if (!response.ok) {
-          response.json().then((data) => {
-            dispatch(addErrors(data.errors.map(err => err.message)));
-            resolve();
-          });
-        } else {
-          resolve();
-        }
+export const updatePageModel = pageModel => dispatch => new Promise((resolve) => {
+  putPageModel(pageModel).then((response) => {
+    if (!response.ok) {
+      response.json().then((data) => {
+        dispatch(addErrors(data.errors.map(err => err.message)));
+        resolve();
       });
     } else {
+      dispatch(addToast(
+        formattedText('app.updated', null, { type: 'page model', code: pageModel.code }),
+        TOAST_SUCCESS,
+      ));
+      gotoRoute(ROUTE_PAGE_MODEL_LIST);
       resolve();
     }
   });
-};
+});
 
-export const createPageModel = pageModel => (dispatch, getState) => {
-  const toSend = pageModel || getFormPageModel(getState());
-  return new Promise((resolve) => {
-    if (toSend) {
-      postPageModel(toSend).then((response) => {
-        if (!response.ok) {
-          response.json().then((data) => {
-            dispatch(addErrors(data.errors.map(err => err.message)));
-            resolve();
-          });
-        } else {
-          resolve();
-        }
+export const createPageModel = pageModel => dispatch => new Promise((resolve) => {
+  postPageModel(pageModel).then((response) => {
+    if (!response.ok) {
+      response.json().then((data) => {
+        dispatch(addErrors(data.errors.map(err => err.message)));
+        resolve();
```

Comment:
you can remove this. You can call resolve after else

Reply:
no I cannot, this one resolves after another promise, while the else is not waiting for that one.